lib/identity_cache/cached/attribute_by_one.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module IdentityCache
+  module Cached
+    class AttributeByOne < Attribute
```

Comment:
Should we rename these to `Attribute::One` and `Attribute::Many`, or maybe `SingleAttribute` and `BatchAttribute`? `*ByOne` and `*ByMulti` seems very fixated on the implementation to me. WDYT?

Reply:
The `By` word is important, because this is referring to the key.  It actually isn't fixated on the caching implementation, it is meant to correspond with the usage. E.g. `cache_attribute :quantity, by: :name` uses Cached::AttributeByOne and `cache_attribute :quantity, by: [:name, :vendor]` uses Cached::AttributeByMulti.  In both cases they are only caching a single attribute, although it wouldn't be hard to support multiple attributes in the future.